src/passes/LogExecution.cpp

```diff
@@ -81,14 +106,40 @@ struct LogExecution : public WalkerPass<PostWalker<LogExecution>> {
 
     import->base = LOGGER;
     curr->addFunction(std::move(import));
+
+    // Reserve all function indices up front for the function names. This is
+    // so that the logged ordinal numbers will match up with the function ordinals.
+    int idx = 0;
+    for (auto& func : curr->functions) {
+       if (func->imported())
+         ++idx;
+    }
+
+    for (auto& func : curr->functions) {
+      if (func->imported())
+        continue;
+
+      Index currentFunctionIndex = (Index)stringToIndex(func->name.toString().c_str());
```

Comment:
```suggestion
      Index currentFunctionIndex = stringToIndex(func->name.toString().c_str());
```

Also, why try to convert function names to indices at all? It would be more robust to avoid trying to interpret the names so that all function names can be treated uniformly.

Reply:
Hey @tlively, it's been a while, though I am now coming back to this functionality, and look to get this landed.

Updated the PR here: the idea here is to ensure that Wasm functions that do not have a debug symbol name string, would be guaranteed to get the log ID that matches its function ordinal. It feels more consistent that way.

Revised the PR to make sure this is the case always, even if some producer might e.g. give debug symbol names to only some of the functions, and not all.